lib/driver.js

```diff
@@ -818,7 +837,20 @@ class XCUITestDriver extends BaseDriver {
               `Try to remove the WebDriverAgentRunner application from the device if it is installed ` +
               `and reboot the device.`;
           }
-          await quitAndUninstall(errorMsg);
+          if (this.usePreinstalledWDA()) {
+            // In case the bundle id process start got failed because of
+            // auth popup in the device. Then, the bundle id process itself started. It is safe to stop it here.
+            await this.mobileKillApp(this.wda.bundleIdForXctest());
+            // Mostly it failed to start the WDA process as no the bundle id
+            // e.g. '<bundle id of WDA> not found on device <udid>'
+            errorMsg =
```

Comment:
do we need to redefine errorMsg here?

Reply:
yea, since existing message explained xcodebuild failure something, which could confuse the user as this use case.

Reply:
but how? It is not used below anywhere (except of the `else` block)

Reply:
Do you mean giving directly or define a new const for `throw new Error(errorMsg);` in L850 instead of using existing `errorMsg`?

Reply:
yes, something like

```
throw new Error(
  `Unable to launch WebDriverAgent because of failure: ${err.message}. ` +
  `Please make sure if the ${this.wda.bundleIdForXctest} exists and it is launchable. ` +
  `${WDA_REAL_DEV_TUTORIAL_URL} may help to complete the preparation.`
)
```

Reply:
got it